src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2452,12 +2437,106 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
+                fixed (byte* inData = &MemoryMarshal.GetReference(bytes))
+                {
+                    int converted = ConvertToBase64Array(outChars, inData, 0, bytes.Length, insertLineBreaks);
+                    Debug.Assert(converted == charLengthRequired);
+                }
+            }
+
+            charsWritten = charLengthRequired;
+            return true;
+        }
+
+        /// <summary>Base64 encodes the bytes from <paramref name="bytes"/> into <paramref name="chars"/>.</summary>
+        /// <param name="bytes">The bytes to encode.</param>
+        /// <param name="chars">The destination buffer large enough to handle the encoded chars.</param>
+        /// <param name="charLengthRequired">The pre-calculated, exact number of chars that will be written.</param>
+        private static unsafe void ToBase64CharsLargeNoLineBreaks(ReadOnlySpan<byte> bytes, Span<char> chars, int charLengthRequired)
+        {
+            // For large enough inputs, it's beneficial to use the vectorized UTF8-based Base64 encoding
+            // and then widen the resulting bytes into chars.
+            Debug.Assert(bytes.Length >= Base64VectorizationLengthThreshold);
+            Debug.Assert(chars.Length >= charLengthRequired);
+
+            // Base64-encode the bytes directly into the destination char buffer (reinterpreted as a byte buffer).
+            OperationStatus status = Base64.EncodeToUtf8(bytes, MemoryMarshal.AsBytes(chars), out _, out int bytesWritten);
+            Debug.Assert(status == OperationStatus.Done && charLengthRequired == bytesWritten);
+
+            // Now widen the ASCII bytes in-place to chars (if the vectorized ASCIIUtility.WidenAsciiToUtf16 is ever updated
+            // to support in-place updates, it should be used here instead). Since the base64 bytes are all valid ASCII, the byte
+            // data is guaranteed to be 1/2 as long as the char data, and we can widen in-place.
+            ref char dest = ref MemoryMarshal.GetReference(chars);
+            ref byte src = ref Unsafe.As<char, byte>(ref dest);
+            ref byte srcBeginning = ref src;
+
+            // We process the bytes/chars from right to left to avoid overwriting the remaining unprocessed data.
+            // The refs start out pointing just past the end of the data, and each iteration of a loop bumps
+            // the refs back the apropriate amount and performs the copy/widening.
+            dest = ref Unsafe.Add(ref dest, charLengthRequired);
+            src = ref Unsafe.Add(ref src, charLengthRequired);
+
+            // Handle 32 bytes at a time.
+            if (Vector256.IsHardwareAccelerated)
+            {
+                ref byte srcBeginningPlus31 = ref Unsafe.Add(ref srcBeginning, 31);
+                while (Unsafe.IsAddressGreaterThan(ref src, ref srcBeginningPlus31))
+                {
+                    src = ref Unsafe.Subtract(ref src, 32);
```

Comment:
Instead of `32` we can use `Vector256<byte>.Count` which is treated by JIT as a const
```suggestion
                    src = ref Unsafe.Subtract(ref src, Vector256<byte>.Count);
```

Reply:
We could, but we'd then need to use Vector256<byte>.Count - 1 instead of 31, and Vector256<byte>.Count / 2 instead of 16. If the type were something larger than byte, I'd agree with you. But as the type doesn't require any additional math, I think it's clearer to just use the constant values.